pkgs/tools/text/gawk/gawkextlib.nix

```diff
@@ -0,0 +1,97 @@
+{ stdenv, recurseIntoAttrs, fetchgit, writeText, pkgconfig, autoreconfHook, autoconf, automake
+, libtool, texinfo, gettext, gawk, rapidjson, gd, shapelib, libharu, lmdb, gmp
+, glibcLocales, mpfr, more, postgresql, hiredis, expat, tre, makeWrapper
+}:
+
+let
+  buildGawkextlibExtension = extension: extraDeps:
+    stdenv.mkDerivation rec {
+      pname = "gawkextlib-${extension}";
+      version = "unstable-2019-11-21";
+
+      src = fetchgit {
+        url = "git://git.code.sf.net/p/gawkextlib/code";
+        rev = "f70f10da2804e4fd0a0bac57736e9c1cf21e345d";
+        sha256 = "0r8fz89n3l4dfszs1980yqj0ah95430lj0y1lb7blfkwxa6c2xik";
+      };
+
+      nativeBuildInputs =
+        [ autoconf automake libtool autoreconfHook pkgconfig texinfo gettext ];
+
+      buildInputs = [ gawk gawkextlib ] ++ extraDeps;
+
+      configureFlags = [
+        "--with-gawkextlib=${gawkextlib}/lib/"
+        "LDFLAGS=-L${gawkextlib}/lib"
+      ];
+
+      postPatch = ''
+        cd ${extension}
+      '';
+      installPhase = ''
+        runHook preInstall
+        mkdir -p $out/lib
+        cp ./.libs/* $out/lib
+        runHook postInstall
+      '';
+
+      # Needed for some tests
+      preCheck = ''
+        AWKLIBPATH=${gawk}/lib/gawk
+      '';
+      setupHook = ./setup-hook.sh;
+
+      doCheck = stdenv.isLinux;
+
+      meta = with stdenv.lib; {
+        homepage = "https://sourceforge.net/projects/gawkextlib/";
+        description = "Dynamically loaded extension libraries for GNU AWK";
+        longDescription = ''
+          The gawkextlib project provides several extension libraries for
+          gawk (GNU AWK), as well as libgawkextlib containing some APIs that
+          are useful for building gawk extension libraries. These libraries
+          enable gawk to process XML data, interact with a PostgreSQL
+          database, use the GD graphics library, and perform unlimited
+          precision MPFR calculations.
+        '';
+        license = licenses.gpl3Plus;
+        platforms = platforms.unix;
+        maintainers = with maintainers; [ tomberek ];
+      };
+    };
+  gawkextlib = (buildGawkextlibExtension "lib" []).overrideAttrs (old: {
+    configureFlags = [];
+    buildInputs = [ gawk ];
+    postInstall = ''
+      cp ../lib/gawkextlib.h $out/lib/.
+    '';
+  });
+in recurseIntoAttrs {
+
+  # callPackage injects extra items into the root attrSet
+  extensions = {
```

Comment:
I still don't get the purpose of this extra nesting.

Reply:
> callPackage injects extra items into the root attrSet

True, is that an issue?

Reply:
I think callPackage puts "override" and "overrideDerivation" into the set. Otherwise i can just import, but then we don't get the callPackage magic. Then if it's an attrSet, it needs to turn into a list somewhere, either in top-level, or once it goes to gawk-with-extensions.nix, either way, the extra items need to be removed.

New version just filters those out at the point of use.